buildenv/jenkins/JenkinsfileBase

```diff
@@ -117,34 +120,52 @@ def setupEnv() {
 def setupParallelEnv() {
 	stage('setupParallelEnv') {
 		timestamps{
-			setupEnv()
+			setupEnv()		
+			
+			//Initialization for things in common between both multiple iterations and many subtests
+			//repeatAmount is the number of times the loops go through
 			def testSubDirs = []
-			def testSubDirSize = 0
+			int repeatAmount = ITERATIONS
 			
-			//For parallel suppose BUILD_LIST is one specific category
-			dir("$WORKSPACE/openjdk-tests/${env.BUILD_LIST}") {
-				testSubDirs = sh(returnStdout: true, script: "ls -d */").trim().tokenize()
-				testSubDirSize = testSubDirs.size()
+			//Setup for option-specific parameters
+			if (repeatAmount == 1){
+				//For parallel suppose BUILD_LIST is one specific category
+				dir("$WORKSPACE/openjdk-tests/${env.BUILD_LIST}") 
+				{
+					testSubDirs = sh(returnStdout: true, script: "ls -d */").trim().tokenize()
+					repeatAmount = testSubDirs.size()
+				}
 			}
-			echo "testSubDirSize is ${testSubDirSize}, testSubDirs is ${testSubDirs}, running test in parallel mode"
+			
+			echo "repeatAmount is ${repeatAmount}, testSubDirs is ${testSubDirs}, running test in parallel mode"
```

Comment:
Please move the echo into if. Otherwise, for ITERATIONS!=1, the output is `testSubDirs is []`